sdks/go/pkg/beam/windowing.go

```diff
@@ -21,21 +21,47 @@ import (
 	"github.com/apache/beam/sdks/go/pkg/beam/internal/errors"
 )
 
+type WindowIntoOption interface {
+	windowIntoOption()
+}
+
+type WindowTrigger struct {
+	Name window.TriggerType
+}
+
+func (t WindowTrigger) windowIntoOption() {}
+
+func (t WindowTrigger) GetName() window.TriggerType {
+	return t.Name
+}
+
 // WindowInto applies the windowing strategy to each element.
-func WindowInto(s Scope, ws *window.Fn, col PCollection) PCollection {
-	return Must(TryWindowInto(s, ws, col))
+func WindowInto(s Scope, ws *window.Fn, col PCollection, opts ...WindowIntoOption) PCollection {
+	return Must(TryWindowInto(s, ws, col, opts...))
 }
 
 // TryWindowInto attempts to insert a WindowInto transform.
-func TryWindowInto(s Scope, ws *window.Fn, col PCollection) (PCollection, error) {
+func TryWindowInto(s Scope, ws *window.Fn, col PCollection, opts ...WindowIntoOption) (PCollection, error) {
 	if !s.IsValid() {
 		return PCollection{}, errors.New("invalid scope")
 	}
 	if !col.IsValid() {
 		return PCollection{}, errors.New("invalid input pcollection")
 	}
+	var edge *graph.MultiEdge
+	for _, opt := range opts {
```

Comment:
Consider just creating a windowing strategy before iterating over the ops with all the reasonable defaults set, and then update that strategy with the whatever is in options. Then we don't end up with a combinatorial explosion when other options are added or with the awkward "if there are no options, do this" length check written below.